sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/mongodb/MongoDbTable.java

```diff
@@ -84,13 +91,41 @@
     return readDocuments.apply(DocumentToRow.withSchema(getSchema()));
   }
 
+  @Override
+  public PCollection<Row> buildIOReader(
+      PBegin begin, BeamSqlTableFilter filters, List<String> fieldNames) {
+    MongoDbIO.Read readBuilder =
+        MongoDbIO.read().withUri(dbUri).withDatabase(dbName).withCollection(dbCollection);
+
+    final FieldAccessDescriptor resolved =
+        FieldAccessDescriptor.withFieldNames(fieldNames)
+            .withOrderByFieldInsertionOrder()
+            .resolve(getSchema());
+    final Schema newSchema = SelectHelpers.getOutputSchema(getSchema(), resolved);
+
+    if (!(filters instanceof DefaultTableFilter)) {
+      throw new RuntimeException("Unimplemented at the moment.");
```

Comment:
I think an `AssertionError` would be more appropriate here, since it should be impossible for this to happen. Also a more descriptive message would be good - maybe "Received a predicate even though predicate push-down is not supported"?

Reply:
I agree that the error message could be clearer here, but a predicate push-down PR for MongoDB is in the works, so this exception won't be there for too long.

Reply:
Well regardless, I think it should be an `AssertionError` to indicate that this is something that we believe to be impossible. Unless I'm misunderstanding and there is some way this could actually happen?

Reply:
Yes, you are correct, this code-path is impossible. I may have misunderstood your first message: do you mean something like `Preconditions.checkArgument` by `AssertionError`?

Reply:
Oh no I meant just throwing `AssertionError` instead of `RuntimeException`: `throw new AssertionError("blah blah")`

something like checkArgument may be a nice way to do the same thing, I'm not sure.

Reply:
I see, will update in a next commit, thanks!